include/seqan3/search/dream_index/interleaved_bloom_filter.hpp

```diff
@@ -599,8 +599,62 @@ class interleaved_bloom_filter<data_layout_mode>::bulk_agent::binning_bitvector
     friend class bulk_agent;
     using sdsl::bit_vector::resize;
     using sdsl::bit_vector::set_int;
+    template <std::integral value_t>
+    friend class counting_vector;
 };
 
 //!\}
 
+template <std::integral value_t>
+class counting_vector : public std::vector<value_t>
+{
+private:
+    using base_t = std::vector<value_t>;
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    counting_vector() = default; //!< Defaulted.
+    counting_vector(counting_vector const &) = default; //!< Defaulted.
+    counting_vector & operator=(counting_vector const &) = default; //!< Defaulted.
+    counting_vector(counting_vector &&) = default; //!< Defaulted.
+    counting_vector & operator=(counting_vector &&) = default; //!< Defaulted.
+    ~counting_vector() = default; //!< Defaulted.
+
+    using base_t::base_t;
+    //!\}
+
+    template <typename t>
+    //!\cond
+        requires std::is_base_of<sdsl::bit_vector, t>::value
+    //!\endcond
+    counting_vector & operator+=(t const & rhs)
+    {
+        assert(size() >= rhs.size()); // The counting vector may be bigger than what we need.
+        size_t bin{0};
+        // Each iteration can handle 64 bits, so we need to iteratate `((rhs.size() + 63) >> 6` masny times
+        for (size_t batch = 0; batch < ((rhs.size() + 63) >> 6); ++batch)
```

Comment:
```suggestion
        // Each iteration can handle 64 bits, so we need to iteratate `((rhs.size() + 63) >> 6` masny times
        for (size_t batch = 0, bin = 0; batch < ((rhs.size() + 63 + 63) >> 6); bin = 64 * batch++)
```

Also add testcase:
0^64 1^64